apk/apk_test.go

```diff
@@ -0,0 +1,116 @@
+package apk
+
+import (
+	"io/ioutil"
+	"log"
+	"os"
+	"path"
+	"testing"
+
+	"github.com/goreleaser/nfpm"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestRunit(t *testing.T) {
+	workDir := path.Join("testdata", "workdir")
+	tempDir, err := ioutil.TempDir(workDir, "test-run")
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer func() {
+		assert.Nil(t, os.RemoveAll(tempDir))
+	}()
+
+	apkFileToCreate := path.Join(tempDir, "apkToCreate.apk")
+
+	err = runit(
+		path.Join("testdata", "files"),
+		path.Join("testdata", "keyfile", "id_rsa"),
+		tempDir,
+		apkFileToCreate)
+
+	assert.Nil(t, err)
+
+	verifyFileSize(t, apkFileToCreate, 1383, 1342, 1379)
+
+	verifyFileSize(t, path.Join(tempDir, "apk_control.tgz"), 304, 300, 305)
+	verifyFileSize(t, path.Join(tempDir, "apk_data.tgz"), 414, 373, 407)
+	verifyFileSize(t, path.Join(tempDir, "apk_signatures.tgz"), 665, 665, 667)
+}
+
+func verifyFileSize(t *testing.T, fileToVerify string, expectedSize, expectedSizeCiMin, expectedSizeCiMax int64) {
+	fi, err := os.Stat(fileToVerify)
+	assert.Nil(t, err)
+	ciEnv := os.Getenv("CI")
+	if ciEnv != "" {
```

Comment:
Why is it that the CI would need a size range?

Reply:
> Why is it that the CI would need a size range?

Only because stuff is broken. :) This whole PR is not ready for prime time, I'm just trying to get the ball rolling forwards.